### PR TITLE
Pizza bombs can no longer be told apart, some pizza improvements

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Pizza bombs are now subtypes of pizza boxes, meaning they can actually blend in now / look / act the same, and can be stacked / have a tag added to them.
Pizza bomb removed out of it's own file, moved into the pizza box file.
50 pc pizza bomb spawner renamed to 33pc pizza bomb spawner (it was 33% chance not 50, and more flavours added to the pile)
Pizza box tags now update on examine, rather than not being visable till being stacked.
You can now write the tag on a pizza box as whatever you want, rather than adding on to it.

## Why It's Good For The Game

Being able to tell pizza bombs apart from pizza boxes suck. Now they blend in well!
Being able to write whatever you want on a pizza box is good
Pizza showing what is written on their tag without having to stack them is good.

## Images of changes

https://cdn.discordapp.com/attachments/145700319819464704/1063922494895042681/Paradise_Station_13_2023-01-14_15-46-17.mp4

## Testing

Confirmed stacking / examining / penning and such worked the same.
Confirmed pizza bombs blended in
Confirmed penning rewrote rather than adding on to.

## Changelog
:cl:
tweak: Pizza bombs blend in better.
tweak: You can now update the tag on a pizza better
add: More pizza on the syndicate base!
fix: Pizza box tags not always showing on examine.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 1]
```diff
@@ -414,3 +418,133 @@
...
+	return ..()
+
+/obj/item/pizzabox/pizza_bomb/update_icon_state()
+	if(primed)
+		return
+	return ..()
+
+/obj/item/pizzabox/pizza_bomb/attack_self(mob/user)
+	if(disarmed)
+		to_chat(user, "<span class='notice'>\The [src] is disarmed.</span>")
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:447]:
```suggestion
		to_chat(user, "<span class='notice'>[src] is disarmed.</span>")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 2]
```diff
@@ -414,3 +418,133 @@
...
+		timer = (input(user, "Set a timer, from one second to ten seconds.", "Timer", "[timer]") as num) SECONDS
+		if(!in_range(src, usr) || issilicon(usr) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || usr.restrained())
+			timer_set = 0
+			name = "pizza box"
+			desc = "A box suited for pizzas."
+			icon_state = "pizzabox1"
+			return
+		timer = clamp(timer, 10, 100)
+		icon_state = "pizzabox1"
+		to_chat(user, "<span class='notice'>You set the timer to [timer / 10] before activating the payload and closing \the [src].")
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:463]:
```suggestion
		to_chat(user, "<span class='notice'>You set the timer to [timer / 10] before activating the payload and closing [src].")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 3]
```diff
@@ -414,3 +418,133 @@
...
+	if(istype(I, /obj/item/wirecutters) && primed)
+		to_chat(user, "<span class='danger'>Oh God, what wire do you cut?!</span>")
+		var/chosen_wire = input(user, "OH GOD OH GOD", "WHAT WIRE?!") in wires
+		if(!in_range(src, usr) || issilicon(usr) || HAS_TRAIT(usr, TRAIT_HANDS_BLOCKED) || usr.restrained())
+			return
+		playsound(src, I.usesound, 50, 1, 1)
+		user.visible_message("<span class='warning'>[user] cuts the [chosen_wire] wire!</span>", "<span class='danger'>You cut the [chosen_wire] wire!</span>")
+		sleep(5)
+		if(chosen_wire == correct_wire)
+			src.audible_message("<span class='warning'>[bicon(src)] \The [src] suddenly stops beeping and seems lifeless.</span>")
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:514]:
```suggestion
			src.audible_message("<span class='warning'>[bicon(src)] [src] suddenly stops beeping and seems lifeless.</span>")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 4]
```diff
@@ -414,3 +418,133 @@
...
+			return
+		else
+			to_chat(user, "<span class='userdanger'>WRONG WIRE!</span>")
+			go_boom()
+			return
+	if(istype(I, /obj/item/wirecutters) && disarmed)
+		if(!in_range(user, src))
+			to_chat(user, "<span class='warning'>You can't see the box well enough to cut the wires out.</span>")
+			return
+		user.visible_message("<span class='notice'>[user] starts removing the payload and wires from \the [src].</span>")
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:530]:
```suggestion
		user.visible_message("<span class='notice'>[user] starts removing the payload and wires from [src].</span>")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 5]
```diff
@@ -414,3 +418,133 @@
...
+		if(do_after(user, 40 * I.toolspeed, target = src))
+			playsound(src, I.usesound, 50, 1, 1)
+			user.unEquip(src)
+			user.visible_message("<span class='notice'>[user] removes the insides of \the [src]!</span>")
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:534]:
```suggestion
			user.visible_message("<span class='notice'>[user] removes the insides of [src]!</span>")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 6]
```diff
@@ -414,3 +418,133 @@
...
+/obj/item/pizzabox/pizza_bomb/cmag_act(mob/user)
+	if(!HAS_TRAIT(src, TRAIT_CMAGGED))
+		to_chat(user, "<span class='notice'>You smear the bananium ooze all over the pizza bomb's internals! You think you smell a bit of tomato sauce.</span>")
+		ADD_TRAIT(src, TRAIT_CMAGGED, CLOWN_EMAG)
+
+/obj/item/pizzabox/pizza_bomb/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/wirecutters) && primed)
+		to_chat(user, "<span class='danger'>Oh God, what wire do you cut?!</span>")
+		var/chosen_wire = input(user, "OH GOD OH GOD", "WHAT WIRE?!") in wires
+		if(!in_range(src, usr) || issilicon(usr) || HAS_TRAIT(usr, TRAIT_HANDS_BLOCKED) || usr.restrained())
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:508]:
```suggestion
		if(!in_range(src, user) || issilicon(usr) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 7]
```diff
@@ -414,3 +418,133 @@
...
+	if(disarmed)
+		to_chat(user, "<span class='notice'>\The [src] is disarmed.</span>")
+		return
+	if(!timer_set)
+		name = "pizza bomb"
+		desc = "It seems inactive."
+		icon_state = "pizzabox_bomb"
+		timer_set = 1
+		timer = (input(user, "Set a timer, from one second to ten seconds.", "Timer", "[timer]") as num) SECONDS
+		if(!in_range(src, usr) || issilicon(usr) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || usr.restrained())
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:455]:
```suggestion
		if(!in_range(src, user) || issilicon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 8]
```diff
@@ -414,3 +418,133 @@
 	. = ..()
 	pizza = new /obj/item/reagent_containers/food/snacks/sliceable/pizza/firecrackerpizza(src)
 	box_tag = "extra spicy pie"
+
+//////////////////////////
+//		Pizza bombs		//
+//////////////////////////
+/obj/item/pizzabox/pizza_bomb
+	var/timer = 10 //Adjustable timer
+	var/timer_set = 0
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:427]: While You're Here, think you could set these to booleans if they're used that way?

**Review by lewcc — COMMENTED** [review]
Quick nits

File: code/modules/food_and_drinks/food/foods/pizza.dm (hunk 8 above)
Comment (Qwertytoforty) [reply code/modules/food_and_drinks/food/foods/pizza.dm]: Done!

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 9]
```diff
@@ -414,3 +418,133 @@
...
+
+/obj/item/pizzabox/pizza_bomb/proc/go_boom()
+	if(disarmed)
+		visible_message("<span class='danger'>[bicon(src)] Sparks briefly jump out of the [correct_wire] wire on [src], but it's disarmed!</span>")
+		return
+	atom_say("Enjoy the pizza!")
+	visible_message("<span class='userdanger'>[src] violently explodes!</span>")
+	message_admins("A pizza bomb set by [key_name_admin(armer)] and opened by [key_name_admin(opener)] has detonated at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[loc.x];Y=[loc.y];Z=[loc.z]'>(JMP)</a>.")
+	log_game("Pizza bomb set by [key_name(armer)] and opened by [key_name(opener)]) detonated at ([loc.x],[loc.y],[loc.z]).")
+	explosion(src.loc,1,2,4,flame_range = 2) //Identical to a minibomb
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/pizza.dm:494]:
```suggestion
	explosion(loc, 1, 2, 4, flame_range = 2) //Identical to a minibomb
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 10]
```diff
@@ -414,3 +418,133 @@
...
+	if(istype(I, /obj/item/wirecutters) && primed)
+		to_chat(user, "<span class='danger'>Oh God, what wire do you cut?!</span>")
+		var/chosen_wire = input(user, "OH GOD OH GOD", "WHAT WIRE?!") in wires
+		if(!in_range(src, user) || issilicon(usr) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
+			return
+		playsound(src, I.usesound, 50, 1, 1)
+		user.visible_message("<span class='warning'>[user] cuts the [chosen_wire] wire!</span>", "<span class='danger'>You cut the [chosen_wire] wire!</span>")
+		sleep(5)
+		if(chosen_wire == correct_wire)
+			src.audible_message("<span class='warning'>[bicon(src)] [src] suddenly stops beeping and seems lifeless.</span>")
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/pizza.dm:514]:
```suggestion
			audible_message("<span class='warning'>[bicon(src)] [src] suddenly stops beeping and seems lifeless.</span>")
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 11]
```diff
@@ -414,3 +418,133 @@
...
+			to_chat(user, "<span class='warning'>You can't see the box well enough to cut the wires out.</span>")
+			return
+		user.visible_message("<span class='notice'>[user] starts removing the payload and wires from [src].</span>")
+		if(do_after(user, 40 * I.toolspeed, target = src))
+			playsound(src, I.usesound, 50, 1, 1)
+			user.unEquip(src)
+			user.visible_message("<span class='notice'>[user] removes the insides of [src]!</span>")
+			var/obj/item/stack/cable_coil/C = new /obj/item/stack/cable_coil(src.loc)
+			C.amount = 3
+			new /obj/item/bombcore/miniature(src.loc)
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/pizza.dm:537]:
```suggestion
			new /obj/item/bombcore/miniature(loc)
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 12]
```diff
@@ -414,3 +418,133 @@
...
+			new /obj/item/pizzabox(src.loc)
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/pizza.dm:538]:
```suggestion
			new /obj/item/pizzabox(loc)
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 13]
```diff
@@ -414,3 +418,133 @@
...
+	..()
+
+/obj/item/pizzabox/pizza_bomb/Initialize(mapload)
+	. = ..()
+	correct_wire = pick(wires)
+	box_tag = "classic pepperoni"
+
+/obj/item/pizzabox/pizza_bomb/autoarm
+	timer_set = TRUE
+	timer = 30 // 3 seconds
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/pizza.dm:550]:
```suggestion
	timer = 3 SECONDS
```

**Review by Sirryan2002 — APPROVED** [review]
this code could really do without using `sleep()` but tbh its kinda out of scope for that change

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 14]
```diff
@@ -251,6 +251,10 @@
 	. = ..()
 	update_appearance(UPDATE_DESC|UPDATE_ICON)
 
+/obj/item/pizzabox/examine(mob/user)
+	update_appearance(UPDATE_DESC|UPDATE_ICON)
+	return ..()
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/food_and_drinks/food/foods/pizza.dm:256]: Needs to fire on examine due to some shenanigans with pre-spawned pizzas and the tags that spawn on them, wasn't able to get a good way to make it work without examine updating it

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**